File: insight-brain-service/src/main/frontend/utility/services/systemConfigurationPropertyService.js
/*
 * Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
 * Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
 * "Sonatype" is a trademark of Sonatype, Inc.
 */
export default
function systemConfigurationPropertyService($http, $rootScope, $q, CLMLocations) {
  return {
    isSuccessMetricsEnabled: isSuccessMetricsEnabled,
    saveSuccessMetricsEnabled: saveSuccessMetricsEnabled,
    checkSuccessMetricsEnabled: checkSuccessMetricsEnabled
  };

  function isSuccessMetricsEnabled() {
    return getSystemConfigurationProperty('SUCCESS_METRICS_ENABLED').then(function(value) {
      return value === 'true';
    });
  }

  function checkSuccessMetricsEnabled() {
    return isSuccessMetricsEnabled().then(function(enabled) {
      if (enabled) {
        return $q.resolve(true);
      }
      else {
        return $q.reject('Success metrics have been disabled by your system administrator.');
      }
    });
  }

  function saveSuccessMetricsEnabled(successMetricsEnabled) {
    var successMetricsProperty = {name: 'SUCCESS_METRICS_ENABLED', value: successMetricsEnabled.toString()};
    return saveSystemConfigurationProperty(successMetricsProperty).then(function(response) {
      $rootScope.$broadcast('successMetricsConfigurationUpdated', response.value === 'true');
      return response;
    });
  }

  function getSystemConfigurationProperty(name) {
    return $http.get(CLMLocations.getSystemConfigurationPropertyUrl(name)).then(function(response) {
      return response.data.value;
    });
  }

  function saveSystemConfigurationProperty(systemConfigurationProperty) {
    return $http.put(CLMLocations.getSystemConfigurationPropertiesUrl(), systemConfigurationProperty).then(function(response) {
      return response.data;
    });
  }
}

systemConfigurationPropertyService.$inject = ['$http', '$rootScope', '$q', 'CLMLocations'];

File: insight-brain-service/src/main/java/com/sonatype/insight/brain/configuration/SystemConfigurationPropertyService.java
/*
 * Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
 * Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
 * "Sonatype" is a trademark of Sonatype, Inc.
 */
package com.sonatype.insight.brain.configuration;

import javax.inject.Inject;
import javax.inject.Named;

import com.sonatype.insight.brain.dataaccess.configuration.SystemConfigurationPropertyDAO;
import com.sonatype.insight.brain.model.configuration.SystemConfigurationProperty;
import com.sonatype.insight.brain.model.security.Permission;
import com.sonatype.insight.brain.security.Authorize;

/**
 * @since 1.33
 */
@Named
public class SystemConfigurationPropertyService
{
  private final SystemConfigurationPropertyDAO systemConfigurationPropertyDAO;

  @Inject
  public SystemConfigurationPropertyService(SystemConfigurationPropertyDAO systemConfigurationPropertyDAO) {
    this.systemConfigurationPropertyDAO = systemConfigurationPropertyDAO;
  }

  public SystemConfigurationProperty getByName(String name) {
    return systemConfigurationPropertyDAO.getByNameNotNull(name);
  }

  @Authorize(permission = Permission.CONFIGURE_SYSTEM)
  public SystemConfigurationProperty update(SystemConfigurationProperty property) {
    systemConfigurationPropertyDAO.update(property);
    return property;
  }
}
